sdk/identity/identity/src/credentials/environmentCredential.ts

```diff
@@ -109,15 +109,15 @@ export class EnvironmentCredential implements TokenCredential {
    * @param scopes - The list of scopes for which the token will have access.
    * @param options - Optional parameters. See {@link GetTokenOptions}.
    */
-  async getToken(
-    scopes: string | string[],
-    options: GetTokenOptions = {}
-  ): Promise<AccessToken | null> {
+  async getToken(scopes: string | string[], options: GetTokenOptions = {}): Promise<AccessToken> {
     return trace("EnvironmentCredential.getToken", options, async (newOptions) => {
       if (this._credential) {
         try {
           const result = await this._credential.getToken(scopes, newOptions);
           logger.getToken.info(formatSuccess(scopes));
+          if (result === null) {
```

Comment:
since `this._credential` is one of ClientSecretCredential, ClientCertificateCredential, or UsernamePasswordCredential, none of which can return `null`, this doesn't ever get called, right?

Can we change the type of `this._credential` to reflect this? Or make it the union of those credential types?